Public/Src/Pips/Dll/Operations/ServiceInfo.cs

```diff
@@ -154,9 +154,9 @@ internal static void InternalSerialize(BuildXLWriter writer, ServiceInfo info)
             writer.Write(info.FinalizationPipIds, (w, v) => WritePipId(w, v));
         }
 
-        private static void WritePipId(BuildXLWriter writer, PipId pipId) => writer.Write(pipId.Value);
+        private static void WritePipId(BuildXLWriter writer, PipId pipId) => writer.WritePipIdValue(pipId.Value);

```

Comment:
>private static void WritePipId(BuildXLWriter writer, PipId pipId) => writer.WritePipIdValue(pipId.Value); [](start = 8, length = 105)

I don't quite understand why we need these `WritePipId` and `ReadPipId`.

Reply:
They need to be remapped when read in onto the current graph based on the pipid the graph gave it when the pip was first added to the graph.  In order to do that, we need to differentiate between writing/reading a uint32 and a pipid.  

---
In reply to: [289581486](https://github.com/microsoft/BuildXL/pull/394#discussion_r289581486) [](ancestors = 289581486)